connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/ConnectAssignorTest.java

```diff
@@ -0,0 +1,138 @@
+                .collect(Collectors.toList());
+        expectedAssignment.get(0).connectors().addAll(Arrays.asList("connector6", "connector9"));
+        expectedAssignment.get(1).connectors().addAll(Arrays.asList("connector7", "connector10"));
+        expectedAssignment.get(2).connectors().addAll(Arrays.asList("connector8"));
+
+        List<String> newConnectors = newConnectors(6, 11);
+        assignor.assignConnectors(existingAssignment, newConnectors);
+        assertEquals(expectedAssignment, existingAssignment);
+    }
+
+    @Test
+    public void testAssignTasksWhenBalanced() {
+        int num = 2;
+        List<WorkerLoad> existingAssignment = IntStream.range(0, 3)
+                .mapToObj(i -> workerLoad("worker" + i, i * num, num, i * num, num))
+                .collect(Collectors.toList());
+
+        List<WorkerLoad> expectedAssignment = existingAssignment.stream()
+                .map(wl -> WorkerLoad.copy(wl.worker(), wl.connectors(), wl.tasks()))
+                .collect(Collectors.toList());
+
+        expectedAssignment.get(0).connectors().addAll(Arrays.asList("connector6", "connector9"));
+        expectedAssignment.get(1).connectors().addAll(Arrays.asList("connector7", "connector10"));
+        expectedAssignment.get(2).connectors().addAll(Arrays.asList("connector8"));
+
+        expectedAssignment.get(0).tasks().addAll(Arrays.asList(new ConnectorTaskId("task", 6), new ConnectorTaskId("task", 9)));
+        expectedAssignment.get(1).tasks().addAll(Arrays.asList(new ConnectorTaskId("task", 7), new ConnectorTaskId("task", 10)));
+        expectedAssignment.get(2).tasks().addAll(Arrays.asList(new ConnectorTaskId("task", 8)));
+
+        List<String> newConnectors = newConnectors(6, 11);
+        assignor.assignConnectors(existingAssignment, newConnectors);
+        List<ConnectorTaskId> newTasks = newTasks(6, 11);
+        assignor.assignTasks(existingAssignment, newTasks);
+        assertEquals(expectedAssignment, existingAssignment);
+    }
+
+    @Test
+    public void testAssignConnectorsWhenImbalanced() {
+        int num = 2;
+        IntUnaryOperator arithmeticProgression = i -> i * ((i - 1) + 2 * num) / 2;
+        List<WorkerLoad> existingAssignment = IntStream.range(0, 3)
+                .mapToObj(i -> workerLoad("worker" + i,
+                        arithmeticProgression.applyAsInt(i), num + i,
+                        arithmeticProgression.applyAsInt(i), num + i))
+                .collect(Collectors.toList());
+        existingAssignment.add(emptyWorkerLoad("worker4"));
+
+        List<String> newConnectors = newConnectors(9, 24);
+        List<ConnectorTaskId> newTasks = newTasks(9, 24);
+        assignor.assignConnectors(existingAssignment, newConnectors);
+        assignor.assignTasks(existingAssignment, newTasks);
+        for (WorkerLoad worker : existingAssignment) {
+            assertEquals(6, worker.connectorsSize());
+            assertEquals(6, worker.tasksSize());
+        }
```

Comment:
Is this all that we can assert here? It doesn't really seem to validate much of the logic by just checking the sizes, does it?

Reply:
The goal of this test is to validate balanced assignment. Thus asserting the size is deterministic and probably sufficient. 

If we agree on the scheduling algorithms additional unit tests will be added (possibly here) and in `WorkerCoordinatorIncrementalTest`

Reply:
This test class, now called `IncrementalCooperativeAssignorTest` has been significantly extended to include a big set of tests for the new assignor. Resolving this comment and will follow up on other test additions.